src/hotspot/share/prims/jvmtiEventController.cpp

```diff
@@ -724,6 +724,7 @@ JvmtiEventControllerPrivate::recompute_enabled() {
     // set global should_post_on_exceptions
     JvmtiExport::set_should_post_on_exceptions((any_env_thread_enabled & SHOULD_POST_ON_EXCEPTIONS_BITS) != 0);
 
+    JvmtiExport::_should_post_allocation_notifications = JvmtiExport::should_post_vm_object_alloc();
```

Comment:
I'm not sure why this flag is needed. It looks like a dup of `JvmtiExport::should_post_vm_object_alloc()`.
Can we just replace it with `JvmtiExport::should_post_vm_object_alloc()`?

Reply:
I don't think we could replace it by function. Also, I think that it is needed later to add SampledObjectAlloc event here. It should consider VM internal object allocations along with all allocations.

Reply:
I was thinking about an offset.
But I've got your plan to use this flag for both `VMObjectAlloc` and `SampledObjectAlloc` event types.